### PR TITLE
use test mode in ResNet predict

### PR DESCRIPTION
**Pull request by mottodora**
Inferring from only one sample with train mode BatchNorm does not work.
I think it is necessary to use `with chainer.using_config('train', False)`.

### CODE REVIEW

File: chainer/links/model/vision/resnet.py [hunk 1]
```diff
@@ -264,13 +265,14 @@ def predict(self, images, oversample=True):
...
         # Use no_backprop_mode to reduce memory consumption
         with function.no_backprop_mode():
-            x = Variable(self.xp.asarray(x))
-            y = self(x, layers=['prob'])['prob']
-            if oversample:
-                n = y.data.shape[0] // 10
-                y_shape = y.data.shape[1:]
-                y = reshape(y, (n, 10) + y_shape)
-                y = sum(y, axis=1) / 10
+            with chainer.using_config('train', False):
```
Comment (mitmul) [inline chainer/links/model/vision/resnet.py]: Both train=False and no_backprop_mode should be specified here like this:

```
with chainer.using_config('train', False), function.no_backprop_mode():
```

**Comment by mitmul** [comment]
@mottodora Could you apply the same change to VGG16Layers and GoogLeNet too?

**Comment by mottodora** [comment]
Yes. I've done.

**Comment by mitmul** [comment]
Jenkins, test this please

**Comment by mitmul** [comment]
LGTM

**Pull request merged**